sdk/Lusid.Sdk.Tests/tutorials/Instruments/InterestRateSwapVariantExamples.cs

```diff
@@ -0,0 +1,161 @@
+    {
+        /// <inheritdoc />
+        protected override void CreateAndUpsertMarketDataToLusid(string scope, ModelSelection.ModelEnum model, LusidInstrument instrument)
+        {
+            // The price of a floating leg is determined by historic resets rates and projected rates.
+            // In this method, we upsert reset rates.
+            // For LUSID to pick up these quotes, we have added a RIC rule to the recipe (see BuildRecipeRequest in TestDataUtilities.cs) 
+            // The RIC rule has a large quote interval, this means that we can use one reset quote for all the resets.
+            // For accurate pricing, one would want to upsert a quote per reset.
+            InterestRateSwap irs = instrument as InterestRateSwap;
+
+            var floatLegs = irs.Legs.OfType<FloatingLeg>();
+            var fixingRefs = floatLegs.Select(floatLeg => floatLeg.LegDefinition.IndexConvention.FixingReference).Distinct().ToList();
+
+            Dictionary<string, UpsertQuoteRequest> resetRequests = new Dictionary<string, UpsertQuoteRequest>();
+
+            // provide resets for each floating leg, with id equal to the fixing reference that the leg will request
+            for (int i = 0; i < fixingRefs.Count; i++)
+            {
+                var fixingRef = fixingRefs[i];
+                TestDataUtilities.BuildQuoteRequest(
+                    resetRequests,
+                    "dummyReset" + i,
+                    fixingRef,
+                    QuoteSeriesId.InstrumentIdTypeEnum.RIC,
+                    0.03m + i/100m,
+                    "InterestRate",
+                    TestDataUtilities.ResetDate,
+                    QuoteSeriesId.QuoteTypeEnum.Price);
+            }
+
+            // provide fx rates for cross-currency swaps
+            var fxRequest = TestDataUtilities.BuildFxRateRequest("USD", "GBP", 0.8m, TestDataUtilities.EffectiveAt, TestDataUtilities.EffectiveAt, useConstantFxRate: true);
+            var quoteRequests = resetRequests.Concat(fxRequest).ToDictionary(d => d.Key, d => d.Value);
+
+            UpsertQuotesResponse upsertResponse = _quotesApi.UpsertQuotes(scope, quoteRequests);
+            Assert.That(upsertResponse.Failed.Count, Is.EqualTo(0));
+            Assert.That(upsertResponse.Values.Count, Is.EqualTo(quoteRequests.Count));
+
+            Dictionary<string, UpsertComplexMarketDataRequest> upsertComplexMarketDataRequest = new Dictionary<string, UpsertComplexMarketDataRequest>();;
+            UpsertStructuredDataResponse upsertComplexMarketDataResponse;
+            // For models requiring discount curves, we upsert them below. ConstantTimeValueOfMoney does not require any discount curves. 
+            if (model != ModelSelection.ModelEnum.ConstantTimeValueOfMoney)
+            {
+                // provide discount/projection curves in USD, for swaps with a 3M floating leg
+                upsertComplexMarketDataRequest.Add("discount_curve_USD",
```

Comment:
You previously argued to "upsert the minimal set". Can this be done here (it looks like you're up'ing all curves)

Reply:
bit messy but done